recipes-core/initrdscripts/files/init-restore-mode.sh

```diff
@@ -86,10 +100,38 @@ mount_nirecovery_usb()
         if [ ! -d $NIRECOVERY_MOUNTPOINT ]; then
             mkdir -p $NIRECOVERY_MOUNTPOINT
         fi
-        mount -o ro,sync,relatime -L NIRECOVERY $NIRECOVERY_MOUNTPOINT &> /dev/null
+        mount -o ro,sync,relatime UUID=$BOOTMEDIA_UUID $NIRECOVERY_MOUNTPOINT &> /dev/null
     fi
 }
 
+mount_payload()
+{
+   if [ -d $NIRECOVERY_MOUNTPOINT/payload ]; then
+      # We are booting from USB so payload will be on same partition as kernel and initrd
+      mount --bind $NIRECOVERY_MOUNTPOINT/payload /payload;
+   else
+      # We used migration ipk, so payload would be in nirootfs
+      ROOTFS_MOUNTPOINT=$(mktemp -d "/var/volatile/rootfs-XXXXXXX")
+      mount -o rw -L nirootfs $ROOTFS_MOUNTPOINT;
+
+      # Payload is in an ipk, extract it
+      mkdir -p $ROOTFS_MOUNTPOINT/payload/extracted
+      pushd $ROOTFS_MOUNTPOINT/payload/extracted
+
+      ar x $ROOTFS_MOUNTPOINT/payload/*.ipk
+      mkdir data
+      tar -xpf data.tar* -C data
+
+      # Ramdisk size is limited, so copy payload to a tmpfs
+      mount -t tmpfs -o size=420m tmpfs /payload
+      cp data/usr/share/nilrt/* /payload
+      mv /payload/*.raucb /payload/niboot.raucb
+      popd
```

Comment:
I think you're still working on non-volatile storage for the this IPK extraction. The contents of `$ROOTFS_MOUNTPOINT/payload/data` are still going to be on nirootfs and contain a copy of the RAUC bundle - which means that we could run out of storage when extracting it.

I was thinking of something more like.
```bash
mount -t tmpfs -o size=420m tmpfs /payload

mkdir /mnt/nirootfs
mount -o ro -L nirootfs /mnt/nirootfs

ar p /mnt/nirootfs/payload/*.ipk data data.tar.xz data.tar.gz | \
    tar -xJp --wildcards '**/*.raucb' -C /payload
find /payload -type f -path '**/*.raucb' -exec mv {} /payload \;
```

That way, we're never writing more to the rootfs partition in the provisioning context. If we do, and we run out of space, it's a shitty customer experience.

Reply:
I intended to remove content from nirootfs after reboot, but looks like I missed it.

> I was thinking of something more like.

I can test it out

Reply:
Ran out of memory doing this.
Added code to remove everything except payload from nirootfs